### ISSUE TITLE
Async generator method is not flagged as a generator

### ISSUE THREAD

**Issue by azz** [eslint/typescript-eslint-parser#297]
**What version of TypeScript are you using?**

2.3.2

**What version of `typescript-eslint-parser` are you using?**

`master`

**What code were you trying to parse?**

```ts
class C {
    async * f() {
        const x = yield* g();
    }
}
```

**What did you expect to happen?**

`generator: true` on `FunctionExpression`.

**What happened?**

`generator: false` on `FunctionExpression`.

Stage 3 async iteration spec: https://tc39.github.io/proposal-async-iteration/#prod-AsyncGeneratorMethod

The TypeScript parser provides an `asteriskToken` on the `MethodDeclaration` node that can be used for this.

**Comment by soda0289** [comment]
This is supported in typescript since 2.3. Must have missed this case. Thanks, will fix.
